### ISSUE TITLE
A bug with multi-threading? Caffe trains with no problem while Darknet quits during training

### ISSUE THREAD

**Issue by Jumabek** [AlexeyAB/darknet#50]
Hi @AlexeyAB , 

I am still having training problem darknet.
I posted it in detail at google groups.

However, since problem is occuring on windows port,
I thought you might be the right person to ask for help.

The gpu training is going smooth on caffe, but not Darknet.
So I have been thinking maybe there is some issue with the software and not the  hardware. 
The copies I produced the error occcured with the YOLOv2 from this repo, nothing I added.
 
Below is my content that I pasted on 
https://groups.google.com/forum/#!topic/darknet/ZCWS6RGMuCM

----------------------------------------------------------------------------------------------------------

So I have been using darknet for last 2 months, everything was just perfect.

However, last week program quits during training saying "darknet.exe is accessing the memory location that does not exist", immediately GPU power, temperature goes down. 
Error persists even when I set subdivisions=batch=1, so I do not think it is because of the memory. Error occurs after training 1K or 2K or 3K iterations.

A) My first doubt is that it might have something to do with core clocks.
1.darknet is setting core clock=1923/1989 MHz, when training with subdivisions=batch=1  while using 1.3GB of GPU memory

  ![](https://lh3.googleusercontent.com/-Yur5gimJNkI/WNxPBbG4A5I/AAAAAAAACCI/4gptxCdNh1EMo2rah-LxBaUGhyrFOBCxACLcB/s1600/darknet_high_core_clock.png)

2. While caffe is setting core clock = 1885 at maximum. while taking 4GB memory for VOC12 classification training on Alexnet (I am just doing this to test GPUs).
 

![](https://lh3.googleusercontent.com/-lreKZ3PQehU/WNxQKdB0buI/AAAAAAAACCU/EfctFuMLVREBtmQiZV-ueJAIftiQe6mzACLcB/s1600/caffe_core_clock.png)

B) My second doubt is that maybe it has something to do with my power Supply (it is not related to darknet, but I think you could be help)

power supply                850 Wt
SSD                            512GB
HDD                            1 TB
CPU                            Core i7-6800K CPU 3.4GHz(12 CPUs), ~3.4GHz
RAM                            16 GB, DDR4

The basis for my doubt is outervision.com power supply calculator recommends me 1200 W PSU even when I excluding GPUs from calculation.
Does that mean I have to buy bigger power supply, but if that is the case why problem does not occur when I train on caffe and occur when I train with darknet?

![](https://lh3.googleusercontent.com/-1KfwmHr6bJ8/WNxWId1rXbI/AAAAAAAACCk/hJq2w4CuaisYCXVDDwLgRPvp7tN0LeGYACLcB/s1600/power_supply_calcualtor.jpg)

 

I also tested my GPU with FUR test and I got above ~7200 score for both of my MSI GTX 1080 Armor GPUs, which seems to be the normal score for 1080.

Any help or suggestions would be appreciated!

Update I:
It is not a temprature issue because when darknet quit, GPU2(darknet was using) did not even hit the 67%.
While at the same time caffe was training on GPU1 with 76% temperature. 

Now I am also starting to think  maybe it is not even a power issue, because when darknet quit, caffe was working fine.

Update II:
I performed the experiment A) by reversing the GPUs as well.
Meaning I trained caffe on GPU2 and darknet on GPU1/
Stil, darknet quits after some time, caffe does not.

**Comment by AlexeyAB** [comment]
@Jumabek Hi,

* Is this issue the same as?: https://github.com/AlexeyAB/darknet/issues/43

* Did you try to train & validate darknet on Linux with the same image-dataset and the same GPU?

* Do you run anything else while training darknet?

* What exactly error is there `program quits during training saying "darknet.exe is accessing the memory location that does not exist"` - preferably screenshot?

**Comment by Jumabek** [comment]
Hi @AlexeyAB,
So until now the problem was occurring when I test or train on wider face detection dataset.
However, yesterday I trained YOLOv2 for VOC by following your instructions,
To my surprise, it is still training without any problem when I came back this morning.
So issue was occurring probably  because of the dataset.

I am not sure exactly what is wrong in wider, I will find it later, but for now I will work on patch based fire detection using YOLO.

My apologies for asking the wrong question. 
And thank you so much for your prompting response and the amazing darknet that you provided us with.
